libraries/AC_Avoidance/AC_Avoid.cpp

```diff
@@ -562,10 +628,6 @@ void AC_Avoid::adjust_velocity_exclusion_circles(float kP, float accel_cmss, Vec
 
     // get desired speed
     const float desired_speed = desired_vel_cms.length();
```

Comment:
if you move the check on desired_speed, you should move the assignement to.
As you can see on line 574 we got a possible return, so it can append that we waste desired_speed creation when we would return.